src/env.rs

```diff
@@ -0,0 +1,40 @@
+use std::env;
+use std::path::{Path, PathBuf};
+
+// Once rust 1.70 is wide-spread (Debian stable), we can use OnceLock from stdlib.
+use once_cell::sync::OnceCell as OnceLock;
+
+static DEFAULT_TEMPDIR: OnceLock<PathBuf> = OnceLock::new();
+
+/// Override the default temporary directory (defaults to [`std::env::temp_dir`]. This function
+/// changes the _global_ default temporary directory for the entire program and should not be called
+/// except in exceptional cases where it's not configured correctly by the platform.
+///
+/// Only the first call to this function will succeed, returning `Ok(path)`. All further calls will
+/// fail with `Err(path)`. In both cases, `path` is the default temporary directory (on success, the
+/// new one; on failure, the existing one).
+pub fn override_temp_dir(path: &Path) -> Result<&'static Path, &'static Path> {
```

Comment:
I considered treating this as a fallback (TMPDIR -> the override -> "/tmp"), but handling all the platform-specific cases got hairy.

I also considered testing if the various temporary directories were writable.... but honestly, I don't want to be in the position of "guessing" where temporary files should be written (although I'll revisit that later if this keeps coming up).

Reply:
I would suggest `Result<(), TempDirAlreadyOverridden>` as the return type.

In the success case there is not point in returning the value just passed in.
In the error case it is problematic that `Path` does not implement `Display`.

Reply:
Docs: Is there a requirement that the directory already exists (I think so) or will it be created (recursively) if non-existent?

Reply:
> In the success case there is not point in returning the value just passed in.

It returns a static reference, which could be convenient if you need to know where the temporary directory is. But... it's inconsistent with `temp_dir`, so it's probably not all that useful.

Reply:
Hm. Ok, I'm going to change it to return `Result<(), PathBuf>`, but I actually don't want the user to `unwrap()` this error, they need to handle it.